hil/client/network.py

```diff
@@ -24,6 +29,22 @@ def create(self, network, owner, access, net_id):
 
             See docs/networks.md for details.
             """
+            bad_chars = self.find_reserved(network)
+            if bool(bad_chars):
+                raise BadArgumentError("Networks may not contain: %s"
+                                       % bad_chars)
+            bad_chars = self.find_reserved(owner)
+            if bool(bad_chars):
+                raise BadArgumentError("Owner may not contain: %s"
+                                       % bad_chars)
```

Comment:
We could probably do a better job of factoring this out. maybe:

```python
check_reserved('Owner', owner)
```

with a suitable definition of `check_reserved`.

We could go a bit further and implement a decorator so we could do something like:

```python
@check_reserved_chars('network', 'owner', 'access', slashes_ok=['net_id'])
def create(self, network, owner, access, net_id):
    # Implementation unchanged from master.
```

It might also make sense to survey the arguments and see what the proportion of arguments to be checked vs. not checked; depending, we may want to specify arguments *not* to check, rather than the other way around.

Reply:
I like the decorator solution, I'm going to work on that.

Reply:
@Izhmash let me know when you do the decorator solution, I'll review this after that.

Reply:
@zenhack how would you relate the arguments in `create()` to the types in the decorator? Since the arguments are passed as `*args` the only way I can see to relate them would be by their order which might not be robust enough.  It would work out if the arguments were changed to be keyworded but that would be editing the original function.

Reply:
I am not sure whether this is right but: what you need is a function that checking the error, for example 

```
def decorator(argument):
    def real_decorator(function):
        def wrapper(*args, **kwargs):
            funny_stuff()
            something_with_argument(argument)
            function(*args, **kwargs)
            more_funny_stuff()
        return wrapper
    return real_decorator
```

> https://stackoverflow.com/questions/5929107/decorators-with-parameters

Reply:
And 
```
@check_reserved_chars(arguments)
class Network(ClientBase):
```
should work so you don't have to put the decorator in every function.

Reply:
@xuhang57 thanks for the input.  The code you presented above definitely fits what I had in mind. The issue lies with relating the non-keyworded arguments in the decorator to the arguments in the methods such as `create()` so we can choose what gets checked.

The only way that I could see to relate (for example) the string `'network'` in the decorator to the `network` passed to the method would be by location since network names have no standard in HIL.  That, then, becomes complicated when considering that some of the objects will allow slashes with the `slashes_ok` flag above.

In the meantime I'm going to work on factoring out my original code and present a solution without decorators.

Reply:
The inspect module may be of interest:

https://docs.python.org/2/library/inspect.html

from there, you can do something like:

```python
argspec = inspect.getargspec(f)
def wrapper(*args):
    for argname, argval in zip(argspec.args, args):
        if argname not in slashes_ok:
            check(argval)
```

Maybe adding some similar logic for kwargs.